libraries/src/Filesystem/File.php

```diff
@@ -174,10 +176,33 @@ public static function copy($src, $dest, $path = null, $useStreams = false)
 				$ret = true;
 			}
 
+			self::invalidateOpcache($dest);
+
 			return $ret;
 		}
 	}
 
+	/**
+	 * Invalidate any opcache for a newly written file immediately, if opcache* functions exist and if this was a PHP file.
+	 *
+	 * @param   string $file  The path to the file just written to, to flush from opcache
+	 *
+	 * @return void
+	 */
+	public static function invalidateOpcache($file)
+	{
+		if (function_exists('opcache_invalidate'))
+		{
+			$info = pathinfo($file);
+
+			if ($info['extension'] === 'php')
```

Comment:
```suggestion
			if (isset($info['extension']) && $info['extension'] === 'php')
```
From [PHP docs](https://www.php.net/manual/en/function.pathinfo.php): "Note: If the path does not have an extension, no extension element will be returned (see second example below)."

Reply:
Done.